extensions/binding/org.eclipse.smarthome.binding.sonyaudio/src/main/java/org/eclipse/smarthome/binding/sonyaudio/handler/SonyAudioHandler.java

```diff
@@ -0,0 +1,536 @@
+                        }
+                    }
+                    break;
+                case CHANNEL_RADIO_SEEK_STATION:
+                    if (command instanceof RefreshType) {
+                        updateState(channelUID, new StringType(""));
+                    }
+                    if (command instanceof StringType) {
+                        switch (((StringType) command).toString()) {
+                            case "fwdSeeking":
+                                connection.radioSeekFwd();
+                                break;
+                            case "bwdSeeking":
+                                connection.radioSeekBwd();
+                                break;
+                        }
+
+                    }
+                    break;
+                default:
+                    logger.error("Channel {} not supported!", id);
+            }
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    public boolean setPowerCommand(Command command) {
+      return ((OnOffType) command) == OnOffType.ON;
+    }
+
+    abstract public String setInputCommand(Command command);
+
+    public int setVolumeCommand(Command command){
+      return ((DecimalType) command).intValue();
+    }
+
+    public boolean setMuteCommand(Command command){
+      return ((OnOffType) command) == OnOffType.ON;
+    }
+
+    public String setSoundFieldCommand(Command command){
+      return ((StringType) command).toString();
+    }
+
+    public boolean setPureDirectCommand(Command command) {
+      return ((OnOffType) command) == OnOffType.ON;
+    }
+
+    public boolean setClearAudioCommand(Command command){
+      return ((OnOffType) command) == OnOffType.ON;
+    }
+
+    public String setRadioStationCommand(int currentRadioStation){
+      return "radio:fm?contentId=" + currentRadioStation;
+    }
+
+    @Override
+    public void initialize() {
+        Configuration config = getThing().getConfiguration();
+        String ipAddress = (String) config.get(SonyAudioBindingConstants.HOST_PARAMETER);
+        String path = (String) config.get(SonyAudioBindingConstants.SCALAR_PATH_PARAMETER);
+        Object port_o = config.get(SonyAudioBindingConstants.SCALAR_PORT_PARAMETER);
+        int port;
+        if (port_o instanceof BigDecimal) {
+            port = ((BigDecimal) port_o).intValue();
+        } else {
+            port = (int) port_o;
+        }
+
+        Object refresh_o = config.get(SonyAudioBindingConstants.REFRESHINTERVAL);
+        int refresh = 0;
+        if (refresh_o instanceof BigDecimal) {
+            refresh = ((BigDecimal) refresh_o).intValue();
+        } else if (refresh_o instanceof Integer) {
+            refresh = (int) refresh_o;
+        }
+
+        try {
+            connection = new SonyAudioConnection(ipAddress, port, path, this);
+
+            connection.connect(scheduler);
```

Comment:
If I see it correctly, this method already tries to establish the socket connection, which means it is potentially long running (e.g. in case of a timeout). You therefore should rather move it to the connectionChecker job (or as this is actually doing the very same, simply delete this line here).